src/terminal/input/terminalInput.hpp

```diff
@@ -87,7 +80,14 @@ namespace Microsoft::Console::VirtualTerminal
         bool _SearchWithModifier(const KeyEvent& keyEvent) const;
 
 
-    public:
+	public:
+        TerminalInput(_In_ std::function<void(std::deque<std::unique_ptr<IInputEvent>>&)> pfn);
+        ~TerminalInput() = default;

```

Comment:
nit: why not just omit the destructor?

Reply:
The ISO C++ guidelines provide relevant guidance in this situation:
*C.21: If you define or =delete any default operation, define or =delete them all*
http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rc-five

Reply:
I understand that the destructor for TerminalInput should be removed.
However, _TermKeyMap has a deleted assignment, so should it not also specify the other 4 or 5 default operations?

Reply:
If one of them is defined,  so should the others. You can mark them as `= default;` or `= delete;` if we aren't doing something special with them, depending on the class.